src/EPR.Calculator.Frontend/Controllers/CalculationRunNameController.cs

```diff
@@ -32,9 +45,33 @@ public IActionResult RunCalculator(InitiateCalculatorRunModel calculationRunMode
             return this.RedirectToAction(ActionNames.RunCalculatorConfirmation);
         }
 
-        public ViewResult Confirmation()
+        public async Task<IActionResult> Confirmation()
         {
-            return this.View(ViewNames.CalculationRunConfirmation);
+            bool confirmationShown = false;
+            try
+            {
+                string runName = this.HttpContext.Session.GetString(SessionConstants.CalculationName) ?? string.Empty;
+
+                var response = await this.GetHttpRequestAsync(runName);
+
+                if (response.IsSuccessStatusCode && response.StatusCode == HttpStatusCode.Accepted)
+                {
+                    confirmationShown = true;
+                }
+            }
+            catch (Exception ex)
+            {
+                this.logger.LogError(ex, "Error during confirmation process.");
+            }
+
+            if (confirmationShown)
+            {
+                return this.View(ViewNames.CalculationRunConfirmation);
+            }
+            else
+            {
```

Comment:
if you remove `confirmationShown`
This can come out of if else loop

Reply:
Removed confirmationShown - @udaydefra Done